compiler/rustc_attr_data_structures/src/attributes.rs

```diff
@@ -120,3 +121,26 @@ impl Deprecation {
         matches!(self.since, DeprecatedSince::RustcVersion(_))
     }
 }
+
+/// Attributes represent parsed, *built in* attributes. That means,
+/// attributes that are not actually ever expanded. They're instead used as markers,
+/// to guide the compilation process in various way in most every stage of the compiler.
+/// These are kept around after the AST, into the HIR and further on.
+///
+/// The word parsed could be a little misleading here, because the parser already parses
+/// attributes early on. However, the result, an [`ast::Attribute`]
+/// is only parsed at a high level, still containing a token stream in many cases. That is
+/// because the structure of the contents varies from attribute to attribute.
+/// With a parsed attribute I mean that each attribute is processed individually into a
+/// final structure, which on-site (the place where the attribute is useful for, think the
+/// the place where `must_use` is checked) little to no extra parsing or validating needs to
+/// happen.
+///
+/// For more docs, look in [`rustc_attr`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_attr/index.html)
+// FIXME(jdonszelmann): rename to AttributeKind once hir::AttributeKind is dissolved
+#[derive(Clone, Debug, HashStable_Generic, Encodable, Decodable)]
+pub enum AttributeKind {
+    // tidy-alphabetical-start
+    DocComment { style: AttrStyle, kind: CommentKind, span: Span, comment: Symbol },
```

Comment:
this will become the type holding all the parsed attributes right now only DocComment, but the next commit adds ~8 more to give an idea of what various parsers look like. In theory, those can be merged individually as well.